engines/stark/ui/menu/saveloadmenu.cpp

```diff
@@ -0,0 +1,256 @@
+			1, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			2, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			3, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			4, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			5, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			6, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			7, _gfx, this));
+	_widgets.push_back(new SaveDataWidget(
+			8, _gfx, this));
+}
+
+void SaveLoadMenuScreen::backHandler() {
+	StarkUserInterface->backPrevScreen();
+}
+
+void SaveLoadMenuScreen::checkError(Common::Error error) {
+	if (error.getCode() != Common::kNoError) {
+		GUI::MessageDialog dialog(error.getDesc());
+		dialog.runModal();
+	}
+}
+
+void SaveMenuScreen::open() {
+	SaveLoadMenuScreen::open();
+	_widgets[kWidgetLoadText]->setVisible(false);
+}
+
+void SaveMenuScreen::onSlotSelected(int slot) {
+	int chapter = StarkGlobal->getCurrentChapter();
+
+	Common::String desc;
+	if (chapter == 0) {
+		desc = "Prologue";
+	} else {
+		desc = Common::String::format("Chapter %d", chapter);
+	}
+
+	g_engine->saveGameState(slot, desc);
+	StarkUserInterface->backPrevScreen();
+}
+
+void LoadMenuScreen::open() {
+	SaveLoadMenuScreen::open();
+	_widgets[kWidgetSaveText]->setVisible(false);
+}
+
+void LoadMenuScreen::onSlotSelected(int slot) {
+	g_engine->loadGameState(slot);
+}
+
+SaveDataWidget::SaveDataWidget(int slot, Gfx::Driver *gfx, SaveLoadMenuScreen *screen) :
+		StaticLocationWidget(nullptr, nullptr, nullptr),
+		_slot(slot),
+		_screen(screen),
+		_thumbWidth(StarkUserInterface->kThumbnailWidth),
+		_thumbHeight(StarkUserInterface->kThumbnailHeight),
+		_texture(gfx->createTexture()),
+		_outline(gfx->createTexture()),
+		_surfaceRenderer(gfx->createSurfaceRenderer()),
+		_textDesc(gfx),
+		_textTime(gfx),
+		_isMouseHovered(false) {
+	// Load the corresponding save slot data
+	Common::String filename = StarkEngine::formatSaveName(ConfMan.getActiveDomainName().c_str(), _slot);
+	Common::InSaveFile *save = g_system->getSavefileManager()->openForLoading(filename);
+	if (save) {
+		SaveMetadata metadata;
+		StateReadStream stream(save);
+		Common::ErrorCode metadataErrorCode = metadata.read(&stream, filename);
+		if (metadataErrorCode != Common::kNoError) {
+			error("Unable to read save metadata with error code %d.", metadataErrorCode);
+		}
+
+		// Obtain the thumbnail
+		Graphics::Surface *thumb = metadata.readGameScreenThumbnail(&stream);
+		_texture->update(thumb);
+		delete thumb;
+
+		// Obtain the text
+		_textDesc.setText(metadata.description);
+		_textDesc.setColor(_textColor);
+		_textDesc.setFont(FontProvider::FontType::kCustomFont, 3);
+
+		_textTime.setText(Common::String::format("%02d:%02d:%02d %02d/%02d/%02d",
+				metadata.saveHour, metadata.saveMinute, metadata.saveSecond,
+				metadata.saveMonth, metadata.saveDay, metadata.saveYear % 100));
+		_textTime.setColor(_textColor);
+		_textTime.setFont(FontProvider::FontType::kCustomFont, 3);
```

Comment:
I think you want FontProvider::kCustomFont instead of FontProvider::FontType::kCustomFont.